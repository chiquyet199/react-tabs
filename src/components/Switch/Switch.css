/* Switch */
.styled-switch {
  height: 32px;
  width: 52px !important;
  border-radius: 16px;
}

.styled-switch > div:first-of-type {
  height: 32px;
  width: 52px;
  border-radius: 16px !important;
  /* colors.steel */
  border: 1px solid #b9c7d6;
}

.styled-switch:not(.active) > div:first-of-type {
  /* colors.white */
  background-color: #ffffff !important;
}

.styled-switch.active > div:first-of-type {
  box-shadow: none;
}

.styled-switch div + div {
  height: 32px;
  width: 32px;
  border-radius: 16px;
  /* colors.steel */
  border: 1px solid #b9c7d6;
  box-shadow: none;
  position: relative;
}

.styled-switch div + div:before {
  content: "\E801";
  font-family: "fontello";
  /* colors.alertRed */
  color: #f16651;
  position: relative;
  font-size: 32px;
  text-align: center;
  line-height: 30px;
  left: -1.5px;
  top: 1.5px;
}

.styled-switch.active div + div {
  margin-left: -31px !important;
}

.styled-switch.active div + div:before {
  content: "\E824";
  /* colors.canopyGreen */
  color: #5ab88e;
}
