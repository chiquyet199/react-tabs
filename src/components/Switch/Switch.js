import React from "react";
import {Switch as SwitchNative} from "react-native";
import {func} from "prop-types";
import {themed} from "../../utils/common";

export const Switch = ({
  disabled = false,
  on = false,
  onChange,
  onChangeCallback,
  theme,
  globaltheme,
  name,
  style,
}) => (
  <SwitchNative
    className={`styled-switch ${on && "active"}`}
    style={style}
    activeTrackColor={theme[globaltheme].switch.accentColor}
    activeThumbColor={theme[globaltheme].switch.baseColor}
    onValueChange={v => {
      onChange(v);
      if (onChangeCallback) {
        onChangeCallback({[`${name}`]: v});
      }
    }}
    disabled={disabled}
    value={on}
  />
);

Switch.propTypes = {
  onChange: func.isRequired,
};

export default themed(Switch);
