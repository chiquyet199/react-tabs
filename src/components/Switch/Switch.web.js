import React from "react";
import {Switch as SwitchNative} from "react-native";
import {func} from "prop-types";
import {themed} from "../../utils/common";
import styles from "./Switch.css";

export const Switch = ({
  disabled = false,
  on = false,
  onChange,
  onChangeCallback,
  theme,
  globaltheme,
  name,
}) => (
  <SwitchNative
    className={`styled-switch ${on && "active"}`}
    activeTrackColor={theme[globaltheme].switch.accentColor}
    activeThumbColor={theme[globaltheme].switch.baseColor}
    onValueChange={v => {
      onChange(v);
      if (onChangeCallback) {
        onChangeCallback({[`${name}`]: v});
      }
    }}
    disabled={disabled}
    value={on}
    {...styles}
  />
);

Switch.propTypes = {
  onChange: func.isRequired,
};

export default themed(Switch);
