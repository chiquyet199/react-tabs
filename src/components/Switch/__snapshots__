/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/Switch/__native_snapshots__