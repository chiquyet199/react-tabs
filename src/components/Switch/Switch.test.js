import React from "react";
import {extend} from "underscore";
import {shallow} from "enzyme";
import {Platform, Switch as SwitchNative} from "react-native";
import {Switch} from "./Switch";

const colors = {
  switchColor: "#5dc471",
  white: "#ffffff",
};
const theme = {
  light: {
    switch: {
      accentColor: colors.switchColor,
      baseColor: colors.white,
    },
  },
};

describe("Switch", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders proper snapshot - " + Platform.OS, () => {
    const tree = getComponent();
    expect(tree).toMatchSnapshot();
  });

  it("renders properly", () => {
    const wrapper = getComponent();
    expect(wrapper.find(SwitchNative)).toHaveLength(1);
  });

  it("has passed props", () => {
    const onChange = jest.fn();
    const wrapper = getComponent({onChange});
    const valueProp = wrapper.find(SwitchNative).props().value;
    const disabledProp = wrapper.find(SwitchNative).props().disabled;
    const activeTrackColor = wrapper
      .find(SwitchNative)
      .prop("activeTrackColor");
    expect(valueProp).toBe(true);
    expect(disabledProp).toBe(false);
    const onValueChangeProp = wrapper.find(SwitchNative).props().onValueChange;
    expect(onValueChangeProp).toBeInstanceOf(Function);
    expect(activeTrackColor).toEqual(colors.switchColor);
  });

  it("calls onChange function when flicked", () => {
    const onChange = jest.fn();
    const wrapper = getComponent({onChange});
    const swn = wrapper.find(SwitchNative);
    const valueProp = swn.props().value;
    swn.props().onValueChange(!valueProp); // trigger
    expect(onChange.mock.calls).toHaveLength(1);
    expect(onChange).toBeCalledWith(false);
    expect(swn.hasClass("active")).toEqual(true);
  });

  it("calls onChangeCallback function if provided with proper object key", () => {
    const fieldName = "test";
    const onChange = () => {};
    const onChangeCallback = jest.fn();
    const comp = getComponent({onChange, onChangeCallback, name: fieldName});
    const swn = comp.find(SwitchNative);
    const valueProp = swn.props().value;
    swn.props().onValueChange(!valueProp); // trigger
    expect(onChangeCallback).toBeCalledWith({[`${fieldName}`]: false});
  });

  it("check class change on false", () => {
    const tree = getComponent({on: false});
    const sw = tree.find(SwitchNative);
    expect(sw.hasClass("active")).toEqual(false);
  });

  const getComponent = props => {
    const parsedProps = extend(getMockProps(), props);

    return shallow(<Switch {...parsedProps} />);
  };
  const getMockProps = () => ({
    on: true,
    onChange: () => {},
    theme,
    globaltheme: "light",
  });
});
