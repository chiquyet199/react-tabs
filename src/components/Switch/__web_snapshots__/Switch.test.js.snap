// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders proper snapshot - web 1`] = `
<Switch
  activeThumbColor="#ffffff"
  activeTrackColor="#5dc471"
  className="styled-switch active"
  disabled={false}
  onValueChange={[Function]}
  style={Object {}}
  thumbColor="#FAFAFA"
  trackColor="#939393"
  value={true}
/>
`;
